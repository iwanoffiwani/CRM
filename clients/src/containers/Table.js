import React, { useState } from 'react'
import PropTypes from 'prop-types' 
import axios from 'axios'
import { connect } from 'react-redux'
import { fetchUpdateOrderList } from '../redux/actions'
import Preloader from '../components/Preloader'
import Layout from './Layout'
import clsx from 'clsx'
import { withStyles, makeStyles } from '@material-ui/core/styles'
import Paper from '@material-ui/core/Paper'
import Table from '@material-ui/core/Table'
import TableBody from '@material-ui/core/TableBody'
import TableCell from '@material-ui/core/TableCell'
import TableHead from '@material-ui/core/TableHead'
import TablePagination from '@material-ui/core/TablePagination'
import TableRow from '@material-ui/core/TableRow'
import MenuItem from '@material-ui/core/MenuItem'
import Select from '@material-ui/core/Select'
import Fab from '@material-ui/core/Fab'
import EditIcon from '@material-ui/icons/Edit'
import Drawer from '@material-ui/core/Drawer'
import Toolbar from '@material-ui/core/Toolbar'
import IconButton from '@material-ui/core/IconButton'
import ChevronLeftIcon from '@material-ui/icons/ChevronLeft'
import Divider from '@material-ui/core/Divider'
import EditOrder from './EditOrder'
import green from '@material-ui/core/colors/green'

const editDrawerWitdth = `75%`

const useStyles = makeStyles(theme => ({
  root: {
    width: '100%',
  },
  wrapper: {
    maxHeight: 790,
    borderRadius: 4,
    overflow: 'auto',
  },
  tabelCell: {
    minWidth: 60
  },
  fab: {
    width: 36,
    height: 36,
    marginTop: `-${theme.spacing(0.8)}px`,
    marginRight: theme.spacing(4)
  },
  editIcon: {
    width: '.8em',
    height: '.8em'
  },
  toolbar: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'flex-end',
    padding: theme.spacing(0, 1),
    ...theme.mixins.toolbar,
  },
  editDrawer: {
    width: editDrawerWitdth
  },
  editDrawerOpen: {
    width: editDrawerWitdth,
    transition: theme.transitions.create('width', {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.enteringScreen
    })
  },
  editDrawerClose: {
    transition: theme.transitions.create('width', {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen,
    }),
    overflowX: 'hidden',
    width: 0
  }
}))

const StyledTableCell = withStyles(theme => ({
  head: {
    backgroundColor: green[500],
    color: theme.palette.common.white,
  },
  body: {
    fontSize: 14,
  },
}))(TableCell)

const mapStateToProps = state => {
  return {
    user: state.authorization.payload,
    columns: state.fields.payload,
    rows: state.orders.payload,
    search: state.search.payload,
    statuses: state.statuses.payload
  }
}

const mapDispatchToProps = dispatch => {
  return {
    update: () => dispatch(fetchUpdateOrderList())
  }
}

export const TableOrders = props => {

  const classes = useStyles()

  const initialState = {
    edit: {
      drawerOpen: false,
      order: false
    },
    pagination: {
      page: 0,
      rowsPerPage: 10
    }
  }

  const [ state, setState ] = useState(initialState)

  const changeStatusHandler = (e, order) => {
    const { 
      data: { 
        login: user 
      } 
    } = props.user // Получаем логин узера для записи в массив изменений

    const {
      _id: orderID,
      status: {
        name: prevStatus
      },
      changes
    } = order // Получаем id заявки, текущий статус заявки и массив изменений

    const {
      id: statusID
    } = e._targetInst.stateNode // Получаем id статуса

    const nextStatus = e.target.value 

    return axios({
        method: 'PATCH',
        url: '/api/orders/',
        params: {
          id: orderID 
        },
        data: {
          status: {
            _id: statusID,
            name: nextStatus // В модели перезапишется статус и _id. Это нужно для последующего редактирования
          },
          changes: [
            ...changes,
            {
              user,
              previousState: { // Свойство должно именоваться именно так, иначе будут серьезные проблемы
                status: prevStatus
              },
              nextState: {
                status: nextStatus
              }
            }
          ]
        },
      })
      .then(() => props.update())
  }

  const openEditDrawerHandler = order => {
    return setState({
      ...state,
      edit: {
        drawerOpen: true,
        order
      }
    })
  }

  const changePageHandler = (event, page)=> {
    return setState({
      ...state,
      pagination: {
        ...state.pagination,
        page
      }
    })
  }

  const changeRowsPerPagehHandler = e => {
    return setState({
      ...state,
      pagination: {
        ...initialState.pagination,
        rowsPerPage: +e.target.value
      }
    })
  }

  const disabledEditDrawerHandler = () => {
    return setState({
      ...state,
      edit: {
        ...initialState.edit
      }
    })
  }

  const otherColumns = 
    props.columns.map((column, index) => 
      <StyledTableCell
        key={index}
        align='right'
        className={classes.tabelCell}
      >{column.name}
      </StyledTableCell>
    )

  if (props.rows.length === 0) 
    return (
      <Preloader />
    )
  else 
    return (
      <Layout>
        <Paper className={classes.root} component='section' elevation={7}>
          <div className={classes.wrapper}>
            <Table>
              <TableHead>
                <TableRow>
                  <StyledTableCell 
                    className={classes.tabelCell}
                  >Имя
                  </StyledTableCell>
                  <StyledTableCell 
                    className={classes.tabelCell}
                  >Статус
                  </StyledTableCell>
                  {otherColumns}
                </TableRow>
              </TableHead>
              <TableBody>
                {(() => {
                  const { page, rowsPerPage } = state.pagination
                  const rows = props.search.length === 0 ? props.rows : props.search
                  
                  return (
                    rows.slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage).map((order, index) => {
                      return (
                        <TableRow hover role='checkbox' tabIndex={-1} key={index}>
                          <TableCell>
                            <Fab 
                              className={classes.fab}
                              size='small'
                              onClick={() => openEditDrawerHandler(order)}
                            >
                              <EditIcon 
                                className={classes.editIcon}
                                aria-label='edit' 
                              />
                            </Fab>
                            {order.name}
                          </TableCell>
                          <TableCell>
                            <Select
                              id='demo-simple-select-outlined'
                              value={order.status.name}
                              variant='outlined'
                              onChange={e => changeStatusHandler(e, order)}
                              fullWidth={true}
                            >{props.statuses.map((status, index) =>
                            <MenuItem 
                              id={status._id}
                              key={index} 
                              value={status.name}
                            >{status.name}</MenuItem>
                            )}
                            </Select>
                          </TableCell>
                          {order.fields.map(field => 
                            <TableCell 
                              key={field._id}
                              align='right'
                            >{field.value}
                            </TableCell>
                          )}
                        </TableRow>
                      )
                    })
                  )
                })()}
              </TableBody>
            </Table>
          </div>
          <TablePagination 
            rowsPerPageOptions={[10, 25, 100]}
            component='div'
            count={props.search.length === 0 ? props.rows.length : props.search.length}
            rowsPerPage={state.pagination.rowsPerPage}
            page={state.pagination.page}
            onChangePage={changePageHandler}
            onChangeRowsPerPage={changeRowsPerPagehHandler}
          />
        </Paper>
        <Drawer 
          open={state.edit.drawerOpen}
          className={clsx(classes.editDrawerOpen, {
            [classes.editDrawerOpen]: state.edit.drawerOpen,
            [classes.editDrawerClose]: !state.edit.drawerOpen,
          })}
          classes={{
            paper: clsx({
              [classes.editDrawerOpen]: state.edit.drawerOpen,
              [classes.editDrawerClose]: !state.edit.drawerOpen,
            }),
          }}
        >
          <Toolbar className={classes.toolbar}>
            <IconButton onClick={disabledEditDrawerHandler}>
              <ChevronLeftIcon />
            </IconButton>
          </Toolbar>
          <Divider />
          <EditOrder 
            editOrder={state.edit.order}
          />
        </Drawer>
      </Layout>
    )
}

TableOrders.propTypes = {
  columns: PropTypes.array.isRequired,
  rows: PropTypes.array.isRequired
}

export default connect(mapStateToProps, mapDispatchToProps)(TableOrders)