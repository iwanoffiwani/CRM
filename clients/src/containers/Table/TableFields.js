import React from 'react'
import { connect } from 'react-redux'
import { withStyles, makeStyles } from '@material-ui/core/styles'
import TableRow from '@material-ui/core/TableRow'
import TableCell from '@material-ui/core/TableCell'
import green from '@material-ui/core/colors/green'

const StyledTableCell = withStyles(theme => ({
  head: {
    backgroundColor: green[500],
    color: theme.palette.common.white,
  },
  body: {
    fontSize: 14,
  },
}))(TableCell)

const useStyles = makeStyles({
  tabelCell: {
    minWidth: 60
  }
})

const TableFields = props => {
  const classes = useStyles()

  const otherFields = 
    props.fields.map(field =>
      <StyledTableCell
        key={field._id}
        align='right'
        className={classes.tabelCell}
      >{field.name}
      </StyledTableCell>
    )

  return (
    <TableRow>
      <StyledTableCell 
        className={classes.tabelCell}
      >Имя
      </StyledTableCell>
      <StyledTableCell 
        className={classes.tabelCell}
      >Статус
      </StyledTableCell>
      {otherFields}
    </TableRow>
  )
}

const mapStateToProps = state => {
  return {
    fields: state.fields.payload
  }
}

export default connect(mapStateToProps)(TableFields)