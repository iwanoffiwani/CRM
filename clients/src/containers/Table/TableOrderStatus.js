import React from 'react'
import { connect } from 'react-redux'
import axios from 'axios'
import { fetchUpdateOrderList } from '../../redux/actions'
import MenuItem from '@material-ui/core/MenuItem'
import Select from '@material-ui/core/Select'

const TableOrderStatus = props => {
  const changeStatusHandler = e => {
    // На прямую получить id заявки нельзя,
    // потому вытаскиваем через переданые свойства
    const stateNode = e._targetInst.stateNode

    const orderID = stateNode.dataset.order
    
    const status = e.target.value

    return axios({
      method: 'PATCH',
      url: '/api/orders/',
      params: {
        id: orderID 
      },
      data: {
        status,
        changes: [
          ...props.order.changes,
          {
            user: props.user,
            previousState: {
              status: props.order.status
            },
            nextState: {
              status
            }
          }
        ]
      },
    })
    .then(() => props.update())
  }

  const otherStatuses = orderID => 
    props.statuses.map(status =>
      <MenuItem 
        id={status._id}
        key={status._id}
        value={status.name}
        data-order={orderID}
      >{status.name}
      </MenuItem>
    )

  return (
    <Select
      id='demo-simple-select-outlined'
      value={props.order.status}
      variant='outlined'
      onChange={changeStatusHandler}
      fullWidth={true}
      inputProps={{ 
        name: 'status'
      }}
    >{otherStatuses(props.order._id)}
    </Select>
  )
}

const mapStateToProps = state => {
  return {
    user: state.auth.payload.user.login,
    statuses: state.statuses.payload
  }
}

const mapDispatchToProps = dispatch => {
  return {
    update: () => dispatch(fetchUpdateOrderList())
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(TableOrderStatus)