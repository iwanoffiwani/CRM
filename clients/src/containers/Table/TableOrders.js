import React from 'react'
import { connect } from 'react-redux'
import { editDrawerOpen } from '../../redux/actions'
import { makeStyles } from '@material-ui/core/styles'
import Fab from '@material-ui/core/Fab'
import EditIcon from '@material-ui/icons/Edit'
import TableCell from '@material-ui/core/TableCell'
import TableRow from '@material-ui/core/TableRow'
import TableOrderStatus from './TableOrderStatus'

const useStyles = makeStyles(theme => ({
  fab: {
    width: 36,
    height: 36,
    marginTop: `-${theme.spacing(0.8)}px`,
    marginRight: theme.spacing(4)
  },
  editIcon: {
    width: '.8em',
    height: '.8em'
  }
}))

const TableOrders = props => {
  const classes = useStyles()

  const editOrderHandler = order => {
    if (!props.edit.drawerOpen) 
      return props.drawerOpen({
        drawerOpen: true,
        order
      })
  }

  const otherOrderFields = fields => 
    fields.map(field => 
      <TableCell 
        key={field._id}
        align='right'
      >{field.value}
      </TableCell>
    )

  return (
    props.orders.map(order => 
      <TableRow hover role='checkbox' tabIndex={-1} key={order._id}>
        <TableCell>
          <Fab 
            className={classes.fab} 
            onClick={() => editOrderHandler(order)}
            size='small'
          >
            <EditIcon 
              className={classes.editIcon}
              aria-label='edit' 
            />
          </Fab>
          {order.name}
        </TableCell>
        <TableCell>
          <TableOrderStatus 
            order={order}
          />
        </TableCell>
        {otherOrderFields(order.fields)}
      </TableRow>
    )
  )
}

const mapStateToProps = state => {
  return {
    edit: state.edit.payload
  }
}

const mapDispatchToProps = dispatch => {
  return {
    drawerOpen: payload => dispatch(editDrawerOpen(payload))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(TableOrders)