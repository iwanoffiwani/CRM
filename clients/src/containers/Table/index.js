import React, { useState } from 'react'
import { connect } from 'react-redux'
import clsx from 'clsx'
import { editDrawerClose } from '../../redux/actions'
import { makeStyles } from '@material-ui/core/styles'
import Layout from '../../components/Layout'
import Paper from '@material-ui/core/Paper'
import Table from '@material-ui/core/Table'
import TableBody from '@material-ui/core/TableBody'
import TableHead from '@material-ui/core/TableHead'
import TablePagination from '@material-ui/core/TablePagination'
import Preloader from '../../components/Preloader'
import Drawer from '@material-ui/core/Drawer'
import Toolbar from '@material-ui/core/Toolbar'
import IconButton from '@material-ui/core/IconButton'
import ChevronLeftIcon from '@material-ui/icons/ChevronLeft'
import Divider from '@material-ui/core/Divider'
import TableFields from './TableFields'
import TableOrders from './TableOrders'
import TableOrderEdit from './TableOrderEdit'

const editDrawerWitdth = `75%`

const useStyles = makeStyles(theme => ({
  root: {
    width: '100%',
  },
  wrapper: {
    maxHeight: 790,
    borderRadius: 4,
    overflow: 'auto',
  },
  toolbar: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'flex-end',
    padding: theme.spacing(0, 1),
    ...theme.mixins.toolbar,
  },
  editDrawer: {
    width: editDrawerWitdth
  },
  editDrawerOpen: {
    width: editDrawerWitdth,
    transition: theme.transitions.create('width', {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.enteringScreen
    })
  },
  editDrawerClose: {
    transition: theme.transitions.create('width', {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen,
    }),
    overflowX: 'hidden',
    width: 0
  }
}))

const OrderTable = props => {
  const classes = useStyles()

  const initialState = { 
    pagination: {
      page: 0,
      rowsPerPage: 10
    }
  }

  const [ state, setState ] = useState(initialState)

  const editDrawerHandler = e => {
    if (props.edit.drawerOpen) 
      return props.drawerClose()
  }

  const changePageHandler = (e, page) => {
    return setState({
      ...state,
      pagination: {
        ...state.pagination,
        page
      }
    })
  }

  const changeRowsPerPageHandler = e => {
    return setState({
      ...state,
      pagination: {
        page: 0,
        rowsPerPage: +e.target.value
      }
    })
  }

  const tableOrders = () => {
    const orders = props.search.length === 0 ? props.orders : props.search

    const slice = {
      begin: state.pagination.page * state.pagination.rowsPerPage,
      end: state.pagination.page * state.pagination.rowsPerPage + state.pagination.rowsPerPage
    }

    // Обрезаем массив для пагинации и последующей передачи его компоненту
    return orders.slice(slice.begin, slice.end)
  }

  if (props.orders.length === 0) 
    return (
      <Preloader />
    )
  else 
    return (
      <Layout>
        <Paper className={classes.root} component='section' elevation={7}>
          <div className={classes.wrapper}>
            <Table stickyHeader aria-label='sticky table'>
              <TableHead>
                <TableFields />
              </TableHead>
              <TableBody>
                <TableOrders 
                  orders={tableOrders()}
                />
              </TableBody>
            </Table>
          </div>
          <TablePagination
            rowsPerPageOptions={[10, 25, 100]}
            component='div'
            count={props.search.length === 0 ? props.orders.length : props.search.length}
            rowsPerPage={state.pagination.rowsPerPage}
            page={state.pagination.page}
            onChangePage={changePageHandler}
            onChangeRowsPerPage={changeRowsPerPageHandler}
          />
        </Paper>
        <Drawer 
          open={props.edit.drawerOpen} 
          className={clsx(classes.editDrawerOpen, {
            [classes.editDrawerOpen]: props.edit.drawerOpen,
            [classes.editDrawerClose]: !props.edit.drawerOpen,
          })}
          classes={{
            paper: clsx({
              [classes.editDrawerOpen]: props.edit.drawerOpen,
              [classes.editDrawerClose]: !props.edit.drawerOpen,
            }),
          }}
          >
          <Toolbar className={classes.toolbar}>
            <IconButton onClick={editDrawerHandler}>
              <ChevronLeftIcon />
            </IconButton>
          </Toolbar>
          <Divider />
          <TableOrderEdit />
        </Drawer>
      </Layout>
    )
}

const mapStateToProps = state => {
  return {
    edit: state.edit.payload,
    orders: state.orders.payload,
    search: state.search.payload
  }
}

const mapDispatchToProps = dispatch => {
  return {
    drawerClose: () => dispatch(editDrawerClose())
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(OrderTable)