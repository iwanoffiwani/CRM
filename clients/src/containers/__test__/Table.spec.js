import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import { TableOrders } from "../Table";
import Fab from "@material-ui/core/Fab";

describe("Tests Table component", () => {
  const mapStateToProps = {
    user: {
      login: "test@test"
    },
    columns: [
      {
        value: "",
        _id: "5dcac8a093dd3b05f8cb0adc",
        name: "Мобильный телефон",
        __v: 0
      },
      {
        value: "",
        _id: "5df9f19e1607be20381256e8",
        name: "Номер паспорта",
        __v: 0
      },
      {
        value: "",
        _id: "5dfa29e98ef6f51a186c395e",
        name: "Домашний тел",
        __v: 0
      },
      {
        value: "",
        _id: "5e00b1bfa63229263405fd21",
        name: "Паспорт",
        __v: 0
      }
    ],
    rows: [
      {
        _id: "5e28371aa7b3b8210c9d38de",
        name: "Новая заявка-35",
        user: "5da47a1353265a14c80ef289",
        fields: [
          {
            value: "+7 666 666 66 66",
            _id: "5dcac8a093dd3b05f8cb0adc",
            name: "Мобильный телефон"
          },
          {
            value: "79 98 666 555",
            _id: "5df9f19e1607be20381256e8",
            name: "Номер паспорта"
          },
          {
            value: "516 69 69 ",
            _id: "5dfa29e98ef6f51a186c395e",
            name: "Домашний тел"
          },
          {
            value: "79 98 654 654",
            _id: "5e00b1bfa63229263405fd21",
            name: "Паспорт"
          }
        ],
        status: {
          _id: "5e28379ffa502c1edc0e768c",
          name: "Статус-4"
        },
        comments: [],
        changes: [
          {
            data: "2020-01-22T11:53:27.209Z",
            _id: "5e2837b7fa502c1edc0e768e",
            user: "test@test",
            previousState: {
              status: "None"
            },
            nextState: {
              status: "Статус-2"
            }
          }
        ]
      }
    ],
    search: [],
    statuses: [
      {
        _id: "5e28378cfa502c1edc0e7689",
        name: "Статус-1",
        __v: 0
      },
      {
        _id: "5e283799fa502c1edc0e768b",
        name: "Статус-2",
        __v: 0
      },
      {
        _id: "5e28379ffa502c1edc0e768c",
        name: "Статус-4",
        __v: 0
      },
      {
        _id: "5e2837a5fa502c1edc0e768d",
        name: "Статус-5",
        __v: 0
      }
    ]
  };

  it("+++ renderer component", () => {
    const wrapper = shallow(<TableOrders {...mapStateToProps} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("+++ simulation of a click event on the order editing button", () => {
    const wrapper = shallow(<TableOrders {...mapStateToProps} />);

    const editButtonWrapper = wrapper.find(Fab).at(0);
    editButtonWrapper.simulate("click");

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
