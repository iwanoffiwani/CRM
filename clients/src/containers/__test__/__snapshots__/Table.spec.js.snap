// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Table component +++ renderer component 1`] = `
<Connect(Layout)>
  <WithStyles(ForwardRef(Paper))
    className="makeStyles-root-1"
    component="section"
    elevation={7}
  >
    <div
      className="makeStyles-wrapper-2"
    >
      <WithStyles(ForwardRef(Table))>
        <WithStyles(ForwardRef(TableHead))>
          <WithStyles(ForwardRef(TableRow))>
            <WithStyles(WithStyles(ForwardRef(TableCell)))
              className="makeStyles-tabelCell-3"
            >
              Имя
            </WithStyles(WithStyles(ForwardRef(TableCell)))>
            <WithStyles(WithStyles(ForwardRef(TableCell)))
              className="makeStyles-tabelCell-3"
            >
              Статус
            </WithStyles(WithStyles(ForwardRef(TableCell)))>
            <WithStyles(WithStyles(ForwardRef(TableCell)))
              align="right"
              className="makeStyles-tabelCell-3"
              key="0"
            >
              Мобильный телефон
            </WithStyles(WithStyles(ForwardRef(TableCell)))>
            <WithStyles(WithStyles(ForwardRef(TableCell)))
              align="right"
              className="makeStyles-tabelCell-3"
              key="1"
            >
              Номер паспорта
            </WithStyles(WithStyles(ForwardRef(TableCell)))>
            <WithStyles(WithStyles(ForwardRef(TableCell)))
              align="right"
              className="makeStyles-tabelCell-3"
              key="2"
            >
              Домашний тел
            </WithStyles(WithStyles(ForwardRef(TableCell)))>
            <WithStyles(WithStyles(ForwardRef(TableCell)))
              align="right"
              className="makeStyles-tabelCell-3"
              key="3"
            >
              Паспорт
            </WithStyles(WithStyles(ForwardRef(TableCell)))>
          </WithStyles(ForwardRef(TableRow))>
        </WithStyles(ForwardRef(TableHead))>
        <WithStyles(ForwardRef(TableBody))>
          <WithStyles(ForwardRef(TableRow))
            hover={true}
            key="0"
            role="checkbox"
            tabIndex={-1}
          >
            <WithStyles(ForwardRef(TableCell))>
              <WithStyles(ForwardRef(Fab))
                className="makeStyles-fab-4"
                onClick={[Function]}
                size="small"
              >
                <EditIcon
                  aria-label="edit"
                  className="makeStyles-editIcon-5"
                />
              </WithStyles(ForwardRef(Fab))>
              Новая заявка-35
            </WithStyles(ForwardRef(TableCell))>
            <WithStyles(ForwardRef(TableCell))>
              <WithStyles(ForwardRef(Select))
                fullWidth={true}
                id="demo-simple-select-outlined"
                onChange={[Function]}
                value="Статус-4"
                variant="outlined"
              >
                <WithStyles(ForwardRef(MenuItem))
                  id="5e28378cfa502c1edc0e7689"
                  key="0"
                  value="Статус-1"
                >
                  Статус-1
                </WithStyles(ForwardRef(MenuItem))>
                <WithStyles(ForwardRef(MenuItem))
                  id="5e283799fa502c1edc0e768b"
                  key="1"
                  value="Статус-2"
                >
                  Статус-2
                </WithStyles(ForwardRef(MenuItem))>
                <WithStyles(ForwardRef(MenuItem))
                  id="5e28379ffa502c1edc0e768c"
                  key="2"
                  value="Статус-4"
                >
                  Статус-4
                </WithStyles(ForwardRef(MenuItem))>
                <WithStyles(ForwardRef(MenuItem))
                  id="5e2837a5fa502c1edc0e768d"
                  key="3"
                  value="Статус-5"
                >
                  Статус-5
                </WithStyles(ForwardRef(MenuItem))>
              </WithStyles(ForwardRef(Select))>
            </WithStyles(ForwardRef(TableCell))>
            <WithStyles(ForwardRef(TableCell))
              align="right"
              key="5dcac8a093dd3b05f8cb0adc"
            >
              +7 666 666 66 66
            </WithStyles(ForwardRef(TableCell))>
            <WithStyles(ForwardRef(TableCell))
              align="right"
              key="5df9f19e1607be20381256e8"
            >
              79 98 666 555
            </WithStyles(ForwardRef(TableCell))>
            <WithStyles(ForwardRef(TableCell))
              align="right"
              key="5dfa29e98ef6f51a186c395e"
            >
              516 69 69 
            </WithStyles(ForwardRef(TableCell))>
            <WithStyles(ForwardRef(TableCell))
              align="right"
              key="5e00b1bfa63229263405fd21"
            >
              79 98 654 654
            </WithStyles(ForwardRef(TableCell))>
          </WithStyles(ForwardRef(TableRow))>
        </WithStyles(ForwardRef(TableBody))>
      </WithStyles(ForwardRef(Table))>
    </div>
    <WithStyles(ForwardRef(TablePagination))
      component="div"
      count={1}
      onChangePage={[Function]}
      onChangeRowsPerPage={[Function]}
      page={0}
      rowsPerPage={10}
      rowsPerPageOptions={
        Array [
          10,
          25,
          100,
        ]
      }
    />
  </WithStyles(ForwardRef(Paper))>
  <WithStyles(ForwardRef(Drawer))
    className="makeStyles-editDrawerOpen-8 makeStyles-editDrawerClose-9"
    classes={
      Object {
        "paper": "makeStyles-editDrawerClose-9",
      }
    }
    open={false}
  >
    <WithStyles(ForwardRef(Toolbar))
      className="makeStyles-toolbar-6"
    >
      <WithStyles(ForwardRef(IconButton))
        onClick={[Function]}
      >
        <ChevronLeftIcon />
      </WithStyles(ForwardRef(IconButton))>
    </WithStyles(ForwardRef(Toolbar))>
    <WithStyles(ForwardRef(Divider)) />
    <Connect(EditOrder)
      editOrder={false}
    />
  </WithStyles(ForwardRef(Drawer))>
</Connect(Layout)>
`;

exports[`Tests Table component +++ simulation of a click event on the order editing button 1`] = `
<Connect(Layout)>
  <WithStyles(ForwardRef(Paper))
    className="makeStyles-root-1"
    component="section"
    elevation={7}
  >
    <div
      className="makeStyles-wrapper-2"
    >
      <WithStyles(ForwardRef(Table))>
        <WithStyles(ForwardRef(TableHead))>
          <WithStyles(ForwardRef(TableRow))>
            <WithStyles(WithStyles(ForwardRef(TableCell)))
              className="makeStyles-tabelCell-3"
            >
              Имя
            </WithStyles(WithStyles(ForwardRef(TableCell)))>
            <WithStyles(WithStyles(ForwardRef(TableCell)))
              className="makeStyles-tabelCell-3"
            >
              Статус
            </WithStyles(WithStyles(ForwardRef(TableCell)))>
            <WithStyles(WithStyles(ForwardRef(TableCell)))
              align="right"
              className="makeStyles-tabelCell-3"
              key="0"
            >
              Мобильный телефон
            </WithStyles(WithStyles(ForwardRef(TableCell)))>
            <WithStyles(WithStyles(ForwardRef(TableCell)))
              align="right"
              className="makeStyles-tabelCell-3"
              key="1"
            >
              Номер паспорта
            </WithStyles(WithStyles(ForwardRef(TableCell)))>
            <WithStyles(WithStyles(ForwardRef(TableCell)))
              align="right"
              className="makeStyles-tabelCell-3"
              key="2"
            >
              Домашний тел
            </WithStyles(WithStyles(ForwardRef(TableCell)))>
            <WithStyles(WithStyles(ForwardRef(TableCell)))
              align="right"
              className="makeStyles-tabelCell-3"
              key="3"
            >
              Паспорт
            </WithStyles(WithStyles(ForwardRef(TableCell)))>
          </WithStyles(ForwardRef(TableRow))>
        </WithStyles(ForwardRef(TableHead))>
        <WithStyles(ForwardRef(TableBody))>
          <WithStyles(ForwardRef(TableRow))
            hover={true}
            key="0"
            role="checkbox"
            tabIndex={-1}
          >
            <WithStyles(ForwardRef(TableCell))>
              <WithStyles(ForwardRef(Fab))
                className="makeStyles-fab-4"
                onClick={[Function]}
                size="small"
              >
                <EditIcon
                  aria-label="edit"
                  className="makeStyles-editIcon-5"
                />
              </WithStyles(ForwardRef(Fab))>
              Новая заявка-35
            </WithStyles(ForwardRef(TableCell))>
            <WithStyles(ForwardRef(TableCell))>
              <WithStyles(ForwardRef(Select))
                fullWidth={true}
                id="demo-simple-select-outlined"
                onChange={[Function]}
                value="Статус-4"
                variant="outlined"
              >
                <WithStyles(ForwardRef(MenuItem))
                  id="5e28378cfa502c1edc0e7689"
                  key="0"
                  value="Статус-1"
                >
                  Статус-1
                </WithStyles(ForwardRef(MenuItem))>
                <WithStyles(ForwardRef(MenuItem))
                  id="5e283799fa502c1edc0e768b"
                  key="1"
                  value="Статус-2"
                >
                  Статус-2
                </WithStyles(ForwardRef(MenuItem))>
                <WithStyles(ForwardRef(MenuItem))
                  id="5e28379ffa502c1edc0e768c"
                  key="2"
                  value="Статус-4"
                >
                  Статус-4
                </WithStyles(ForwardRef(MenuItem))>
                <WithStyles(ForwardRef(MenuItem))
                  id="5e2837a5fa502c1edc0e768d"
                  key="3"
                  value="Статус-5"
                >
                  Статус-5
                </WithStyles(ForwardRef(MenuItem))>
              </WithStyles(ForwardRef(Select))>
            </WithStyles(ForwardRef(TableCell))>
            <WithStyles(ForwardRef(TableCell))
              align="right"
              key="5dcac8a093dd3b05f8cb0adc"
            >
              +7 666 666 66 66
            </WithStyles(ForwardRef(TableCell))>
            <WithStyles(ForwardRef(TableCell))
              align="right"
              key="5df9f19e1607be20381256e8"
            >
              79 98 666 555
            </WithStyles(ForwardRef(TableCell))>
            <WithStyles(ForwardRef(TableCell))
              align="right"
              key="5dfa29e98ef6f51a186c395e"
            >
              516 69 69 
            </WithStyles(ForwardRef(TableCell))>
            <WithStyles(ForwardRef(TableCell))
              align="right"
              key="5e00b1bfa63229263405fd21"
            >
              79 98 654 654
            </WithStyles(ForwardRef(TableCell))>
          </WithStyles(ForwardRef(TableRow))>
        </WithStyles(ForwardRef(TableBody))>
      </WithStyles(ForwardRef(Table))>
    </div>
    <WithStyles(ForwardRef(TablePagination))
      component="div"
      count={1}
      onChangePage={[Function]}
      onChangeRowsPerPage={[Function]}
      page={0}
      rowsPerPage={10}
      rowsPerPageOptions={
        Array [
          10,
          25,
          100,
        ]
      }
    />
  </WithStyles(ForwardRef(Paper))>
  <WithStyles(ForwardRef(Drawer))
    className="makeStyles-editDrawerOpen-8 makeStyles-editDrawerOpen-8"
    classes={
      Object {
        "paper": "makeStyles-editDrawerOpen-8",
      }
    }
    open={true}
  >
    <WithStyles(ForwardRef(Toolbar))
      className="makeStyles-toolbar-6"
    >
      <WithStyles(ForwardRef(IconButton))
        onClick={[Function]}
      >
        <ChevronLeftIcon />
      </WithStyles(ForwardRef(IconButton))>
    </WithStyles(ForwardRef(Toolbar))>
    <WithStyles(ForwardRef(Divider)) />
    <Connect(EditOrder)
      editOrder={
        Object {
          "_id": "5e28371aa7b3b8210c9d38de",
          "changes": Array [
            Object {
              "_id": "5e2837b7fa502c1edc0e768e",
              "data": "2020-01-22T11:53:27.209Z",
              "nextState": Object {
                "status": "Статус-2",
              },
              "previousState": Object {
                "status": "None",
              },
              "user": "test@test",
            },
          ],
          "comments": Array [],
          "fields": Array [
            Object {
              "_id": "5dcac8a093dd3b05f8cb0adc",
              "name": "Мобильный телефон",
              "value": "+7 666 666 66 66",
            },
            Object {
              "_id": "5df9f19e1607be20381256e8",
              "name": "Номер паспорта",
              "value": "79 98 666 555",
            },
            Object {
              "_id": "5dfa29e98ef6f51a186c395e",
              "name": "Домашний тел",
              "value": "516 69 69 ",
            },
            Object {
              "_id": "5e00b1bfa63229263405fd21",
              "name": "Паспорт",
              "value": "79 98 654 654",
            },
          ],
          "name": "Новая заявка-35",
          "status": Object {
            "_id": "5e28379ffa502c1edc0e768c",
            "name": "Статус-4",
          },
          "user": "5da47a1353265a14c80ef289",
        }
      }
    />
  </WithStyles(ForwardRef(Drawer))>
</Connect(Layout)>
`;
